### PR TITLE
Added loading resource white and black lists

### PR DESCRIPTION
**Pull request by maZahaca**
Added ability to set white and black lists for loading resources to decrease internet traffic.

#53

### CODE REVIEW

**Comment by maZahaca** [comment]
@jifeon for review

**Comment by maZahaca** [comment]
@jifeon for review

**Comment by maZahaca** [comment]
@jifeon for review

File: lib/PhantomEnvironment.js [hunk 1]
```diff
@@ -349,10 +353,12 @@ PhantomEnvironment.prototype = _.create(Environment.prototype, /**@lends Phantom
             deferred.resolve();
         }
         // resourceRequestStarted has a special function...
-        else if (type === 'resourceRequestStarted') {
+        else if (type === 'resourceRequested') {
             var args = [].slice.call(arguments);
-            args = args.slice(1, args.length - 1); // callback OR callback with args
-            this._page.onResourceRequested.apply(this._page, args);
+            args = args.slice(3, args.length);
```
Comment (jifeon) [inline lib/PhantomEnvironment.js]: you don't need second argument in this case

File: lib/PhantomEnvironment.js [hunk 2]
```diff
@@ -390,6 +396,25 @@ PhantomEnvironment.prototype = _.create(Environment.prototype, /**@lends Phantom
             this._on('navigationRequested', function (url) {
                 debug('Navigation to %s', url);
             }),
+            this._on('resourceRequested', /* @covignore */ function (requestData, request, allowedUrls, blockedUrls) {
```
Comment (jifeon) [inline lib/PhantomEnvironment.js]: why it's covignored?

File: lib/PhantomEnvironment.js [hunk 3]
```diff
@@ -390,6 +396,25 @@ PhantomEnvironment.prototype = _.create(Environment.prototype, /**@lends Phantom
...
+                var url = requestData.url;
+                var hasAllowedUrls = Array.isArray(allowedUrls) && allowedUrls.length > 0;
+                var hasBlockedUrls = Array.isArray(blockedUrls) && blockedUrls.length > 0;
+                var allowed = !hasAllowedUrls || allowedUrls.reduce(function (lastValue, urlPattern) {
```
Comment (jifeon) [inline lib/PhantomEnvironment.js]: you can use [some](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/some) or [every](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/every)

**Comment by jifeon** [comment]
@maZahaca reviewed

**Comment by jifeon** [comment]
@maZahaca reviewed

**Comment by jifeon** [comment]
@maZahaca reviewed

**Comment by jifeon** [comment]
do not forget about documentation

File: lib/PhantomEnvironment.js (hunk 2 above)
Comment (maZahaca) [inline lib/PhantomEnvironment.js]: This function will be executed inside Phantom

**Comment by maZahaca** [comment]
Docs will come in separate PR, it's already done in it.

**Comment by jifeon** [comment]
Now it looks cool!

**Pull request merged**